assets/js/custom/Program.js

```diff
@@ -59,6 +70,36 @@ export const Program = function (projectId, projectName, userRole, myProgram, st
     )
   })
 
+  function steal (url, buttonId, spinnerId, iconId, isWebView = false) {
+    const button = document.getElementById(buttonId)
+    const loadingSpinner = document.getElementById(spinnerId)
+    const icon = document.getElementById(iconId)
+
+    button.disabled = true
+    icon.classList.add('d-none')
+    loadingSpinner.classList.remove('d-none')
+    loadingSpinner.classList.add('d-inline-block')
+    showSnackbar('#share-snackbar', 'Stealing Project...')
+    new ApiFetch(url).generateAuthenticatedFetch()
+      .then(function (response) {
+        if (response.ok) {
+          showSnackbar('#share-snackbar', 'The project has been stolen')
+          return response.blob()
+        }
+      })
+      .catch(() => {
+        // UX: Tell the use that something went wrong
+        showSnackbar('#share-snackbar', 'Error while stealing the project')
+      })
+      .finally(() => {
+        // UX: Reset the button to further indicate the successful download
+        resetDownloadButtonIcon(icon, loadingSpinner)
+        // Performance: Keep the button disabled to prevent spamming the download button
+        setTimeout(() => {
+          button.disabled = false
+        }, 15000)
```

Comment:
pretty sure you copied code you don't need ;)
In your future implementations, try to take a look at the existing solutions to get your first input, but always consider doing the solution in a better way - the project is full of legacy code, and therefore copy - paste may not always be the best solution